### ISSUE TITLE
WSL version 2.0.12 makes $XDG_RUNTIME_DIR not user accessible when guiApplications=false

### ISSUE THREAD

**Issue by elsaco** [microsoft/WSL#10846]
### Windows Version

10.0.19045.3693

### WSL Version

20.12.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.133.1-microsoft-standard-WSL2

### Distro Version

Ubuntu 23.10

### Repro Steps

Set `guiApplications = false` and restart WSL

### Expected Behavior

$XDG_RUNTIME_DIR user accessible

### Actual Behavior

```
elsaco@mantic:~$ ls -lnd $XDG_RUNTIME_DIR
drwx------ 2 0 0 4096 Nov  6 19:10 /run/user/1000/
elsaco@mantic:~$ ls -la $XDG_RUNTIME_DIR
ls: cannot open directory '/run/user/1000/': Permission denied
```

It breaks systemd:

```
elsaco@mantic:~$ systemctl --user status
Failed to connect to bus: Permission denied
```

From debug console:

`WSL (2) ERROR: ConfigHandleInteropMessage:595: getenv(XDG_RUNTIME_DIR) failed`

**Comment by fknittel** [comment]
I've found a work-around. Once per wsl distribution and user you need to fix the permissions of the directory used to mount the ephemeral XDG_RUNTIME directory /mnt/wslg/run/user/<user-id>

Example:

```
$ id -u
1000
$ sudo chown 1000:1000 /mnt/wslg/run/user/1000

```
After that, reboot your wsl and everything should be fine:

```
wsl.exe --reboot
```

Does that work for you @elsaco?

In my opinion, `/mnt/wslg/run/user` shouldn't be bind-mounted to `/run/user` for `guiApplications=false`, because `/mnt/wslg` is not what `/init` expects it to be (initially doesn't exist and gets created on `/`).

**Comment by sgraf812** [comment]
This is a serious issue for me as well. Apparently I need to run @fknittel's workaround after every update.

**Comment by jeffrson** [comment]
For me, /run/user is empty, so "1000" does not exist there at all, which breaks some applications because nevertheless XDG_RUNTIME_DIR is set to /run/user/1000/

**Comment by crlmrlck** [comment]
> I've found a work-around. Once per wsl distribution and user you need to fix the permissions of the directory used to mount the ephemeral XDG_RUNTIME directory /mnt/wslg/run/user/
> 
> Example:
> 
> ```
> $ id -u
> 1000
> $ sudo chown 1000:1000 /mnt/wslg/run/user/1000
> ```
> 
> After that, reboot your wsl and everything should be fine:
> 
> ```
> wsl.exe --shutdown
> ```
> 
> Does that work for you [@elsaco](https://github.com/elsaco)?
> 
> In my opinion, `/mnt/wslg/run/user` shouldn't be bind-mounted to `/run/user` for `guiApplications=false`, because `/mnt/wslg` is not what `/init` expects it to be (initially doesn't exist and gets created on `/`).

I was having this issue after updating to the latest version of WSL a few weeks ago. This fix still works as of 2025.12.03 on version 2.6.1.0

**Comment by trianglesis** [comment]
Why is it closed?
The issue is still there, latest WSL version, OL9.5 distro.


```
WSL version: 2.7.3.0
Kernel version: 6.6.114.1-1
WSLg version: 1.0.73
MSRDC version: 1.2.6676
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26100.1-240331-1435.ge-release
Windows version: 10.0.26200.8457
```
